website/static/js/share/results.js

```diff
@@ -1,126 +1,252 @@
-                                ]);
-                            }())))
+                        m('.pull-right',
+                          {style: {'text-align': 'right'}},
+                          m.component(Subjects, params)
+                        )
+                    )
                 ]),
                 m('br'),
-                m('br'),
-                m('div', [
-                    m('span', 'Released on ' + new $osf.FormattableDate(result.dateUpdated).local),
-                    m('span.pull-right', [
-                        m('img', {src: self.vm.ProviderMap[result.source].favicon, style: {width: '16px', height: '16px'}}),
-                        ' ',
-                        m('a', {onclick: function() {utils.updateFilter(self.vm, 'source:' + result.source);}}, self.vm.ProviderMap[result.source].long_name)
-                    ])
-                ])
+                m.component(Footer, params)
             ]),
             m('hr')
         ]);
-    };
-
-    // Uncomment for infinite scrolling!
-    // $(window).scroll(function() {
-    //     if  ($(window).scrollTop() === $(document).height() - $(window).height()){
-    //         utils.loadMore(self.vm);
-    //     }
-    // });
+
+    }
+};
+
+var TitleBar = {
+    view: function(ctrl, params) {
+        var result = params.result;
+        return m('span', {}, [
+            m('a[href=]' + result.canonicalUri + ']', ((result.title || 'No title provided'))),
+            m('br'),
+            m.component(Description, params)
+        ]);
+    }
+};
+
+/* Render the description of a single result. Will highlight the matched text */
+var Description = {
+    view: function(ctrl, params) {
+        var result = params.result;
+        if ((result.description || '').length > 350) {
+                return m('p.readable.pointer', {
+                    onclick:function(){
+                        result.showAll = result.showAll ? false : true;
+                        }
+                    },
+                    result.showAll ? result.description : $.trim(result.description.substring(0, 350)) + '...'
+                );
+            }
+            return m('p.readable', result.description);
+    }
+};
+
+var Contributors = {
+    view: function(ctrl, params) {
+        var result = params.result;
+        var contributorViews = $.map(result.contributors, function(contributor, i) {
+                return m.component(Contributor, $.extend({contributor: contributor, index: i}, params));
+            });
+
+        return m('span.pull-left', {style: {'text-align': 'left'}},
+            result.showAllContrib || result.contributors.length < 8 ?
+                contributorViews :
+                m('span', [
+                    contributorViews.slice(0, 7),
+                    m('br'),
+                    m('a', {onclick: function(){result.showAllContrib = result.showAllContrib ? false : true;}}, 'See All')
```

Comment:
See https://github.com/CenterForOpenScience/osf.io/pull/3500/files#r36471509 ; I think the principle applies here, too.